src/components/FloatingFlow.vue

```diff
@@ -0,0 +1,202 @@
+<template>
+  <el-container style="height:100%;background:white;">
+    <el-header style="text-align: left; font-size: 14px;padding:0" height="40px" class="dialog-header">
+      <DialogToolbarContent :activeId="activeDockedId" :dialogTitles="dockedArray"
+        :showIcons="entries[findIndexOfId(activeDockedId)].mode!=='main'" @maximise="dockedMaximise"
+        @minimise="dockedMinimise" @close="dockedClose" @titleClicked="dockedTitleClicked"/>       
+    </el-header>
+    <el-main class="dialog-main">
+      <div style="width:100%;height:100%;position:relative;overflow:hidden;">
+        <FloatingDialog v-for="item in entries" :entry="item" :index="item.id"
+          :key="item.id" v-on:mousedown.native="dialogClicked(item.id)"
+          @maximise="dialogMaximise(item.id)" @minimise="dialogMinimise(item.id)" 
+          @close="dialogClose(item.id)" @onActionClick="onActionClick"/>
+      </div>
+    </el-main>
+  </el-container>
+</template>
+
+<script>
+/* eslint-disable no-alert, no-console */
+import DialogToolbarContent from './DialogToolbarContent';
+import FloatingDialog from './FloatingDialog';
+import Vue from "vue";
+import {
+  Container,
+  Header,
+  Main
+} from "element-ui";
+Vue.use(Container);
+Vue.use(Header);
+Vue.use(Main);
+
+export default {
+  name: "FloatingFlow",
+  components: {
+    DialogToolbarContent,
+    FloatingDialog
+  },
+  methods: {
+    onActionClick:function(action) {
+      if (action) {
+        if (action.type == "URL") {
+          window.open(action.resource,'_blank');
+        } else {
+          let newId = this.createNewEntry(action);
+          this.bringDialogToFront(newId);
+        }
+      }
+    },
+    createNewEntry: function(data) {
+      let newEntry = {};
+      newEntry.resource = data.resource;
+      newEntry.type = data.type;
+      newEntry.mode = "normal";
+      if (data.plotType)
+        newEntry.plotType = data.plotType;
+      newEntry.id = ++this.currentCount;
+      newEntry.zIndex = ++this.zIndex;
+      this.entries.push(newEntry);
+      return newEntry.id;
+    },
+    findIndexOfId: function(id) {
```

Comment:
Can the position in entries array not be their 'id'?

Reply:
Not in this case, id is unique while posiiton changed when an item is removed.

Reply:
I'm still not sure I get it,  but I saw somewhere in a different file along the lines of `id == -1`. So I can see how it is used as more than just an index in an array.

Reply:
It is just a way for me to make sure each entry has an identifier that is not going to be changed through out its life time.